DOC-1718: RS Connect entry page (#2232)
RedisLabs/redislabs-docs: content/rs/databases/connect/_index.md

]
---

> After you have [Set up a cluster]({{< relref "/rs/clusters/new-cluster-setup" >}}) and [created a Redis database]({{< relref "/rs/databases/create" >}}), connect to your database to test your connection.

Reviewer: ```suggestion
After you have [Set up a cluster]({{< relref "/rs/clusters/new-cluster-setup" >}}) and [created a Redis database]({{< relref "/rs/databases/create" >}}), you can connect to your database.
```

> After you have [Set up a cluster]({{< relref "/rs/clusters/new-cluster-setup" >}}) and [created a Redis database]({{< relref "/rs/databases/create" >}}), connect to your database to test your connection.

Reviewer: ```suggestion
After you have [Set up a cluster]({{< relref "/rs/clusters/new-cluster-setup" >}}) and [created a Redis database]({{< relref "/rs/databases/create" >}}), you can connect to your database.
```

> To connect to your database, you need the database endpoint, which includes the cluster name (FQDN) and the database port. Select the Configuration tab in the database screen to find the database endpoint.

Reviewer: ```suggestion
To connect to your database, you need the database endpoint, which includes the cluster name (FQDN) and the database port. 

Select the **Configuration** tab in the database screen to find the database endpoint.
```

Reviewer: I felt like the action set was a bit buried here; putting it on a new line might help.

- [RedisInsight](https://redis.com/redis-enterprise/redis-insight/)

## redis-cli {#using-rediscli}

The [`redis-cli`]({{<relref "/rs/references/cli-utilities/redis-cli/">}}) utility is installed when you install Redis.  It provides a command-line interface that lets you work with your database using core [Redis commands](https://redis.io/commands/).

`redis-cli` is located in `/opt/redislabs/bin/` on a node with Redis Enterprise installed. To connect to the database from a node in the cluster, run `redis-cli` with the database port.

```sh
$ sudo /opt/redislabs/bin/redis-cli -p <port>
```

To connect using `redis-cli` from outside of the cluster, run `redis-cli` with the database hostname and port.

```sh
$ sudo /opt/redislabs/bin/redis-cli -h <host> -p <port>
```

## Redis client{#using-redis-client}

Different programming languages use different connection clients to interact with Redis databases, and each client has its own syntax and installation process. For help with a specific client, see the client's documentation.

See the [client list](https://redis.io/docs/clients/) to view all Redis clients by language.

{{< note >}}

> You cannot use client libraries to configure Redis Enterprise Software.  Instead, use:

Reviewer: ```suggestion
You can't use client libraries to configure Redis Enterprise Software.  Instead, use:
```

- The Redis Software [admin console]({{< relref "/rs/installing-upgrading/get-started-redis-enterprise-software.md" >}})
- The [REST API]({{<relref "/rs/references/rest-api">}})
- Command-line utilities, such as [`rladmin`]({{<relref "/rs/references/cli-utilities/rladmin">}})
{{< /note >}}

### Code example (Python)

A simple python application can also connect to the database.

{{< note >}}
The following section assumes you already have Python
and redis-py (python library for connecting to Redis) configured.
You can find the instructions to configure redis-py on the [github page for redis-py](https://github.com/redis/redis-py).
{{< /note >}}

1. Create a new file called `redis_test.py`:

    ```python
    import redis

    r = redis.StrictRedis(host='<host>', port=<port>)
    print ("set key1 123")
    print (r.set('key1', '123'))
    print ("get key1")
    print(r.get('key1'))
    ```

>     Replace <host> and <port> with the hostname and port for your database.

Reviewer: ```suggestion
    Replace `<host>` and `<port>` with the hostname and port for your database.
```
